pkg/controller/validation/account_validation_controller_test.go

```diff
@@ -0,0 +1,304 @@
+			if err := MoveAccount(tt.args.account, tt.args.client, tt.args.targetOU, tt.args.dryRun); (err != nil) != tt.wantErr {
+				t.Errorf("MoveAccount() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestValidateAccount_Reconcile(t *testing.T) {
+	err := apis.AddToScheme(scheme.Scheme)
+	if err != nil {
+		fmt.Printf("failed adding to scheme in account_validation_controller_test.go")
+	}
+	ctrl := gomock.NewController(t)
+	newBuilder := func(ctrl *gomock.Controller) awsclient.IBuilder {
+		mockClient := mock.NewMockClient(ctrl)
+		mockBuilder := mock.NewMockIBuilder(ctrl)
+		mockBuilder.EXPECT().GetClient(gomock.Any(), gomock.Any(), gomock.Any()).Return(mockClient, nil)
+		return mockBuilder
+	}
+	type fields struct {
+		Client           client.Client
+		scheme           *runtime.Scheme
+		awsClientBuilder awsclient.IBuilder
+	}
+	type args struct {
+		request reconcile.Request
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    reconcile.Result
+		wantErr bool
+	}{
+		{name: "Will not attempt to reconcile a CCS account.", fields: fields{
+			Client: fake.NewFakeClient([]runtime.Object{
+				&awsv1alpha1.Account{
+					TypeMeta: v1.TypeMeta{
+						Kind:       "Account",
+						APIVersion: "v1alpha1",
+					},
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "test",
+						Namespace: "default",
+					},
+					Spec: awsv1alpha1.AccountSpec{
+						BYOC: true,
+					},
+				},
+			}...),
+			scheme:           scheme.Scheme,
+			awsClientBuilder: newBuilder(ctrl),
+		}, args: args{
+			request: reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: "default",
+					Name:      "test",
+				},
+			},
+		}, want: reconcile.Result{Requeue: false}, wantErr: false},
+		{name: "Will not attempt to reconcile a claimed account.", fields: fields{
+			Client: fake.NewFakeClient([]runtime.Object{
+				&awsv1alpha1.Account{
+					TypeMeta: v1.TypeMeta{
+						Kind:       "Account",
+						APIVersion: "v1alpha1",
+					},
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "test",
+						Namespace: "default",
+					},
+					Status: awsv1alpha1.AccountStatus{
+						Claimed: true,
+					},
+				},
+			}...),
+			scheme:           scheme.Scheme,
+			awsClientBuilder: newBuilder(ctrl),
+		}, args: args{
+			request: reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: "default",
+					Name:      "test",
+				},
+			},
+		}, want: reconcile.Result{Requeue: false}, wantErr: false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &ValidateAccount{
+				Client:           tt.fields.Client,
+				scheme:           tt.fields.scheme,
+				awsClientBuilder: tt.fields.awsClientBuilder,
+			}
+			got, err := r.Reconcile(tt.args.request)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ValidateAccount.Reconcile() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
I would recommend using go-cmp instead of DeepEqual as it is a better alternative. You can read more here: https://pkg.go.dev/github.com/google/go-cmp/cmp

Reply:
I think the reflect, just gets autogenerated by gotests, so I will change it.